backend/src/main/java/com/example/subjecthub/controller/SubjectServiceController.java

```diff
@@ -101,7 +107,13 @@ public SubjectComment getComment(
         @PathVariable Long subjectId,
         @PathVariable Long commentId
     ){
-        return subjectCommentRepository.findBySubject_IdAndId(subjectId,commentId);
+        SubjectComment result;
+        try {
+            result = subjectCommentRepository.findBySubject_IdAndId(subjectId, commentId);
+        }catch(NullPointerException|ClassCastException e){
+            throw new SubjectHubException("Specified comment id: "+commentId+", not found for subject id: "+subjectId);
+        }
+        return result;
```

Comment:
You're not calling a method on a possibly null object here, so there's no need for the try except block.

Reply:
fixed